### DISCUSSION TITLE
DC QT5 launches slower in GTK based DE

### DISCUSSION THREAD

**Discussion by advice2020** [doublecmd/doublecmd#1471] [General]
-Debian based Linux distro / GTK based DE
-DC 1.2.0 alpha QT5 (revision 11186) portable
(but happens in every single QT5 version I have tried: 1.0.11 from my repo, 1.2.0 Appimages, standalone, etc.)

For future proofing reasons, I am trying to see how smooth it would be to use DC QT5 in a GTK based DE.  In doing so I notice that the QT5 version takes a bit longer to load up then the GTK2 version which is very fast.
May sound foolish to some, but when it comes to programs such as file managers I want them to load up as quickly as possible.  The GTK2 version does just that, it loads just as fast as the default FM that comes with my DE, lets just say it takes 1 second.  The QT5 version takes about 5-6, there is a noticeable pause that occurs, I sometimes question if I actually launched it or not, and then it appears.
I have had two people (one Debian based and I think one Arch based) mention that this does NOT occur for them and that the QT5 version loads just as fast, so it is not an issue with DC itself.
I don't really want to get into hardware specs, but I don't think that it has anything to do with this either, because the GTK2 version loads up fast.  I have also tested out the opposite, using a QT based DE, and the DC QT5 version loads up fast in QT DE as well.
So it appears that it only does NOT work with the setup that I actually want to use:)

I was wondering if anyone knows a reason/solution for why this could be happening or could at least verify this for me whether this happens on your GTK based DE machine or not.

Thank You to anyone who reads this and for any info shared

I will post some of the things I have tried below if anyone is interested.

**Comment by advice2020** [comment]
Non of the following tests worked
-Heard that it might have something to do with gtk2 based themes.
I was originally using "qt5-gtk2-platformtheme" but tried "qt5ct" so I could choose a non gtk2 based theme.

-I tried using the same machine with a few different distros that include a GTK based DE, I tried three different ones that included XFCE, I tried the portable & Appimage QT versions and oddly non of them were able to open up the QT5 versions at all.  Only the GTK2 versions worked, so that was a lot of installing / time for nothing :)

-I tried installing a fresh new version of my distro and went right into testing DC, thinking maybe a package that I installed may have been causing the problem, but nope.

-I saw some people mentioning to install the following package
`sudo apt-get install appmenu-gtk2-module`
then reboot

-As mentioned I tried QT based DE and it DC QT5, this was the only thing that worked, but not what I am looking to use.

If anyone has any more ideas please let me know, even if it is something that I already tried above, with better instructions on how to test.

**Comment by JonnyTech** [comment]
I can confirm. I too run DC Qt build on XFCE. It always takes at least 5 seconds to start.

Just tested with the latest release AppImages:

doublecmd-gtk-latest-x86_64.AppImage < 2 seconds
doublecmd-qt-latest-x86_64.AppImage > 5 seconds

**Reply by j2969719** [reply]
to be fair, you are like comparing a toolkit from the early 2000s with a toolkit from the 2010s. personally doubt that a potential port to gtk3 and etc with its bells and whistles, would launch faster and consume less resources even compared to qt5, not to mention gtk2.

**Comment by advice2020** [comment]
@JonnyTech
Thank you very much for trying this out and posting your results.
You are the only one that seems to be getting these same results that I am.
Yeah 5-6 seconds or more does not make any sense.
I have some information I will post below that may interest you.

@j2969719 
Interesting way of looking at it, I was thinking the opposite, that a newer more modern toolkit would generate faster results, but I do understand what you are saying.
Based on the results that have been shared, GTK2 does always seem to load faster.
QT5 usually loads slightly slower, so this does make sense, however the results that I am getting (and JonnyTech) I believe are more then this, I have info that I will share below.

**Comment by advice2020** [comment]
@Anyone
I was finally able to test out DC QT5 in other distro/DEs then the one I am currently using and they all loaded fine. The GTK2 version still loads faster which is a common result seen, but QT5 result was not much / a perfectly acceptable amount of difference when used in my tests with other distro/DEs. Say GTK2 is 1 second, the QT5 is about 2-3.
So the 5-6 second that I am experiencing in my current distro is strange.

So it does not appear to be hardware related, not DC itself, not toolkit (there is a slight difference between toolkit speeds but not the same as what I am experiencing) and not DE related. I even tried other DEs offered by my distro and they worked as well.
As mentioned I also tested the reverse, QT based DE and both DC GTK & QT loaded fine as well.
So not sure what is causing this.
If anyone else can share results
or if anyone can think of anything else to try please let me know
Thank You

**Comment by advice2020** [comment]
Still curious about this one, if anyone has any ideas.
In the future I plan on testing this out again.

**Comment by alexx2000** [comment]
https://github.com/doublecmd/doublecmd/issues/2297#issuecomment-2887674534

**Comment by advice2020** [comment]
Alexander!!!!!!
I was able to test out the updates you made to how icons are handled during launch and there is a noticeable improvement to the launch speed of the QT5 version.
Thank you for making these updates, well worth the effort.

**Comment by advice2020** [comment]
Had some more observations / test results to share.
So this had to do with how DC was accessing icon sets (which I mentioned in the past something seemed strange with icons when running DC from the terminal), but this must also be something QT related as well, because the GTK2 version before and after this update is still faster.

In the past when I mentioned about GTK2 version being noticeably faster compared to QT5, it was explained that it was because QT5 toolkit was newer which meant it contained more coding/features/etc?

Nice that that timer was added because helps with not having to sit here and verbally count with each test :)
I just realized that this timer also works in the GTK2 version (which I have just been using for some comparison tests).
My QT5 standalone version is about `Creating PixmapManager done 700`
QT5 AppImage version is about `Creating PixmapManager done 750`
but
My GTK2 version is about `Creating PixmapManager done 100`
and this is the AppImage version
So when it comes to accessing these icon sets, the GTK2 version is about 7 times faster.
I imagine it also handles other parts of the launch process faster as well.

But this new update definitely makes my QT5 version launch much quicker then before.
Also I think the QT upper left corner of screen artifacts I am experiencing (mentioned in Discussion post 1503) are also tricking my mind a little to make it appear slightly slower then it actually is.
The user in the other post linked to this one mentioned his QT used to be about 10 seconds, but then recently went up to 20 seconds! These times he experienced were way over what I was experiencing, I wonder how this update improved his results. 

Anyways, just some observations / more info about the subject I thought I would share.
Thank you again for making this update, really great.

**Comment by advice2020** [comment]
Version: 1.2.0 alpha
Revision: 12229
Commit: be4128e1d
Build date: 2025/05/30
Lazarus: 3.6.0.0
FPC: 3.2.2
Platform: x86_64-Linux-qt5
Widgetset library: Qt 5.15.8

I saw in the snapshot changelog that there were a few icon related updates a few weeks ago.
I was not sure specifically what these updates were in reference to but I thought I would recompile just to test if had any affect on launch speed.
For me no changes, icons still about 710, but again I do not know if these updates had anything to do with launch speed, just sharing my test results.
